File: GATEONE DAY 6/PYTHON/AgeCalculator.py
from datetime import datetime

def calculate_age(date_of_birth):
 
    dob = datetime.strptime(date_of_birth, '%d-%m-%Y')

    current_date = datetime.now()
    
    age = current_date.year - dob.year - ((current_date.month, current_date.day) < (dob.month, dob.day))
    
    return age

def main()
    dob_input = input("Enter your date of birth (DD-MM-YYYY): ")
    
    age = calculate_age(dob_input)
    
    print("Age:", age, "years")

if _name_ == "_main":
    main()

File: GATEONE DAY 6/JAVA/AgeCalculator.java
import java.time.LocalDate;
import java.time.Period;
import java.time.format.DateTimeFormatter;
import java.util.Scanner;

public class AgeCalculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter your date of birth (DD-MM-YYYY): ");
        String dobString = scanner.nextLine();

        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
        LocalDate dob = LocalDate.parse(dobString, formatter);

        LocalDate currentDate = LocalDate.now();

        Period period = Period.between(dob, currentDate);

        int years = period.getYears();
        int months = period.getMonths();
        int days = period.getDays();

        System.out.println("Age: " + years);

       
    }
}